app/policies/configuration/agent_policy.rb

```diff
@@ -1,31 +1,51 @@
 class Configuration::AgentPolicy
   def initialize(context, agent)
     @current_agent = context.agent
-    @current_territory = context.territory
     @agent = agent
-    @access_rights = @current_agent.access_rights_for_territory(@current_territory)
   end
 
-  def territorial_admin?
-    @current_agent.territorial_admin_in?(@current_territory)
+  def self.allowed_to_manage_agents_in?(territory, agent)
```

Comment:
j’ai gardé ce mécanisme d’itération sur les `agent_territories.any? { current_agent_has_sufficient_rights_in_territory(_1) }` 

on pourrait aussi l’écrire sous la forme `agent_territories.intersection(territories_where_current_agent_has_sufficient_rights).any?` comme tu le suggérais @victormours 

Il m’a semblé que c’était à peu près équivalent en lisibilité mais je n’ai pas du tout un avis fort la dessus